codalab/worker/file_util.py

```diff
@@ -10,10 +10,15 @@
 
 BINARY_PLACEHOLDER = '<binary>'
 NONE_PLACEHOLDER = '<none>'
+GIT_PATTERN = '.git'
 
 
 def tar_gzip_directory(
-    directory_path, follow_symlinks=False, exclude_patterns=[], exclude_names=[]
+    directory_path,
+    follow_symlinks=False,
+    exclude_patterns=[],
+    exclude_names=[],
+    ignore_file_name='.gitignore',
```

Comment:
Maybe this should be `None` just to maintain backward compatibility?

Reply:
@percyliang: Regarding your comments about backward compatibility, I prefer respecting `.gitignore` as the default, since I think most users do not want to see binary files and extraneous files in CodaLab. Additionally, I can add the optional flag to `cl upload` to give users the freedom to turn it off and print a message based on the value of the flag.